crowbar_framework/app/models/api/error.rb

```diff
@@ -14,7 +14,18 @@
 # limitations under the License.
 #
 
-module Api
-  class Errors
-  end
+class Api::Error < ActiveRecord::Base
+  belongs_to :crowbar
+
+  validates :error,
+    presence: true
+
+  validates :message,
+    presence: true
+
+  validates :code,
+    inclusion: {
+      in: Rack::Utils::HTTP_STATUS_CODES.keys.concat([nil]),
```

Comment:
I wouldn't restrict to only http codes here, as we may want to save zypper codes, shell return codes etc.